storage/watchable_store_test.go

```diff
@@ -222,3 +234,95 @@ func TestUnsafeAddWatching(t *testing.T) {
 		}
 	}
 }
+
+func TestMapWatchingToEvents(t *testing.T) {
```

Comment:
This tests seems to be verbose. Can we populate a map and an array and them just test the func `newWatchingToEventMap`? 

We do not need to construct watchableStore.